### ISSUE TITLE
Zombie Burners are kind of dumb

### ISSUE THREAD

**Issue by anoobindisguise** [CleverRaven/Cataclysm-DDA#62252]
### Is your feature request related to a problem? Please describe.

Zombie Burners are dead flamethrower soldiers. Their tank is described as leaking, and it in fact has a limited capacity of napalm it can leak before it stops leaking entirely. In addition to this, they have a rare chance to explode in a lethal conflagration when killed. 

This obviously doesn't make a whole lot of sense. Why do they explode? Does "Blob Magic" ignite the napalm when they die? Napalm tanks exploding is a hollywood/videogame fabrication first of all, it doesn't really conform to real life. Secondly, there is a good chance the napalm tank is entirely empty by the time they would die, but they still explode anyways. Third, the napalm tanks only start leaking once the zombies are in your reality bubble, when in reality their tanks should have emptied within 10 minutes of them resurrecting as a zombie.

Them exploding is very much a "gotcha" mechanic, the player has no way to reasonably expect that they will explode into flames when killed because of how improbable and unrealistic it is. And one explosion can very easily kill a player in melee especially if they are new enough to the game that they don't know that these enemies can erupt into flames on death.

### Solution you would like.

1. Zombie burners don't leak napalm when moving.
2. They don't explode when killed.

### Describe alternatives you have considered.

Remove zombie burners.

**Comment by Kywi001** [comment]
I agree with the "they shouldn't explode when killed in melee" part,

The leaking part, it really depends on where it's leaking, it can leak from the top part or the bottom part,

And the part where they should have a chance to explode, is when a tracer round or hot shrapels hit them and tore the container open igniting the fuel inside (according to qoura),

All in all, it needs rebalancing
1. Melee shouldn't make them explode.
2. Range (excluding Bows and Crossbows) should give them a chance to explode even without killing them (ofcourse assuming the sharpnel hits the tank).
3. If they started to leak, they should have a chance to drop none to almost empty canister (instead of 600 to 200, change it to 600 to 0).
4. If they exploded, the canister they drop should have 0 napalm left together with a really broken flamethrower.

**Comment by AcidAntOnAMinefield** [comment]
They shouldn't even have a flamethrower to begin with, honestly. They're dead, not feral. Zombies shouldn't be "holding" guns or weapons of any kind. When they died, they didn't die holding the gun close to their chest, afraid someone would steal it: they most likely dropped it when they hit the floor. When they got up, they didn't just magically pick it up.

Plus, if they are holding a flamethrower, then how can they grab you in melee? They need the two hands to hold the weapon, so for them to grab the player/target they need to have free hands. 

IMO, burners shouldn't explode in any potential way, period. Giving them "limbs" to hit like (ie "hit the tank or w/e) we do for NPCs for example will become an incessant pool of RNG bs where the player almost never hits the torso or the head, but the NPC using projectiles *always* hits the head/torso with full accuracy. (Meaning that some NPCs can outright kill you with 2 or 3 9x19 shots, whereas the player needs 2 full magazines to kill them.) This would lead to questionable situations where, say, a "10% chance to hit the tank and the zombie goes boom" becomes a "the game decides you aren't allowed to continue playing 90% of the time" scenario.

**Comment by Night-Pryanik** [comment]
> Why do they explode? Does "Blob Magic" ignite the napalm when they die?

Their flamethrowers had pilot lights turned on when they died, and are still working when they became zombies. That's why they are illuminated at night, if you didn't notice, by the way. With a conjunction of leaking tanks that should justify why they are exploding sometimes. And also this is the reason of why it doesn't matter if they are killed in melee or by ranged weapons.

Also, what's "dumb" with burners? They _may be_ unbalanced (I personally entirely disagree with this), but they behave more or less like other zombified soldiers, so I don't understand why are you calling them dumb.

**Comment by Night-Pryanik** [comment]
> Zombies shouldn't be "holding" guns or weapons of any kind. When they died, they didn't die holding the gun close to their chest

You're right, they don't hold anything in their hands. Flamethrowers are freely hanging out on their straps/slings.

**Comment by Zireael07** [comment]
> Zombies shouldn't be "holding" guns or weapons of any kind. 

Counter argument: we already have zombie soldiers and zombie cops using guns and no one voiced such a complaint... (which personally I think doesn't make sense, why shouldn't they? they can still use their arms)

**Comment by AcidAntOnAMinefield** [comment]
> > Zombies shouldn't be "holding" guns or weapons of any kind.
> 
> Counter argument: we already have zombie soldiers and zombie cops using guns and no one voiced such a complaint... (which personally I think doesn't make sense, why shouldn't they? they can still use their arms)

I've never seen a Zombie Soldier or Zombie Cop using guns. I've seen the Feral counterparts/equivalents doing that - but they are Ferals, not Zombies. And Ferals by themselves are a whole other issue in terms of how accurate they still are and they fact they don't suffer any penalties the player does.

And sure, it's "hanging from their strap/sling." Cool. It still is one of those old "now you die because reasons" situations and that really isn't a good explanation for it though imo.

**Comment by robotic-gamer** [comment]
easiest fix I can think of is to make it so that all  zombies that "explode" when killed very clearly state this property in their description you see when looking at them.

**Comment by Kywi001** [comment]
Some flamethrowers are connected via hose (or something) and their tanks are strapped like a backpack, assuming this, they can still have the flamethrower albeit, just hanging from the said hose, assuming worse case scenario, they should/could accidentally go off by just walking as the said end point is being dragged.

On another side, if its one of those flamethrowers where the gas tank is loaded into the thrower, we can assume that it is on a sling and said gas tank are badly damaged that if the zombie's body dropped on top of it, it might go off.

To be honest, this zombies are greatly helpful if you know what you're doing, and the easiest fix is just adding a clause on it's description that the flamethrower might explode upon death due to a badly damaged flamethrower gas canister.

**Comment by Ilysen** [comment]
I think these comments are missing some main points in the OP. None of them address the issue that the tanks are damaged to the point of leaking on burners, and yet supposedly still have enough fuel left to explode days, weeks, or even months after the zombie died and was reanimated.

It's not like something is gonna be stopping those leaks; it doesn't matter if the pilot light was on when they died or not, physics is still physics. It's not going to erupt into a massive fireball if you kill a zombie and they somehow land on top of their tank in the way because if the tank is dripping constantly, it should be empty. There's even a comment in the code about this - monmove.cpp, line 1832:

```cpp
                    // TODO: remove MF_DRIPS_NAPALM flag since no more napalm in tank
                    // Not possible for now since flag check is done on type, not individual monster
```

The math can lay this out for us here. Zombie burners start with 1000 units of napalm, and have a 10% chance per move to drip 50 units of that; we can average that out to 5 units per move. If we're generous and assumes that your average wandering zombie randomly stumbles one tile every 10 turns while idle, we can thusly assume that every idle turn, a burner will drip out 0.5 units of napalm. Statistically speaking, it'd take roughly 2000 turns to deplete the tank of a wandering burner. That's 33.3 minutes, or less than an hour. Yikes!

I actually went and tested this to make sure my data was right. Giving ourselves debug mutations to make ourselves imperceptible and then spawning our boi in an empty field to wander around, his tank was depleted after 30 minutes of waiting in place, even without any monsters in the way to distract him; it drained all passively as he wandered around.

![image](https://user-images.githubusercontent.com/47678781/202886121-3da9bd51-68fe-4a02-9b80-29b6bc9e28b7.png)

Each burner still has a 10% chance to explode - when I repeated this test by spawning 25 burners and *then* waiting for all of their tanks to drain, four of them still exploded on death despite having completely empty tanks. I agree with the verisimilitude problem in the OP here - the problem isn't that the zombies can explode, for me, as long as it's telegraphed sufficiently; the problem is that their tanks are ruptured and should be empty by now if they've been dead for so long, and even when their tanks *are* empty and they literally leave behind no napalm as they move, they can still explode in a huge fireball when they die from even something like bleeding out or whatever. This doesn't even affect their flamethrower drops, either! Their flamethrowers will still have a random amount of fuel in them regardless of how quickly or how late you kill them, and if we want to assume that the flamethrower tanks are stable enough to maintain their fuel, then why have them drip napalm or explode at all? It just doesn't make sense to me, and I think it's a point that the other comments here aren't responding to.

**Comment by AcidAntOnAMinefield** [comment]
> None of them address the issue that the tanks are damaged to the point of leaking on burners, and yet supposedly still have enough fuel left to explode days, weeks, or even months after the zombie died and was reanimated.

> the problem isn't that the zombies can explode, for me, as long as it's telegraphed sufficiently;


The problem *is* that they can explode, telegraphed or not. The point of the original post was outlined in the first few lines of the second paragraph, with:  "why do they explode, is it blob magic?". The whole "it should only explode when it's not empty" logic is 100% flawed. So, the answer is "yes, it is blob magic", because the stupid "shoot tank with boolet and boolet make tank go boom hehe" logic is purely hollywood logic - it doesn't work that way in real life. In fact, it even doesn't work like that if someone gets shot in the head and falls onto a damaged (or otherwise) tank of fuel - that's even sillier. 


IMO, If it's supposed to be based on the selective realism most of the creatures are based off of, then at the very least the explosion part shouldn't happen. If you want to base it in full realism, most of these zombies with abilities related to their professions make absolutely zero sense and would need to be reworked.

**Comment by Night-Pryanik** [comment]
Right now the problem of burners exploding and dropping their weapons with some amount of napalm even if their tanks are (supposedly) empty is entirely technical. AFAIK there's no way currently to check for monster's ammo before triggering its death effect (I mean jsonified death effect) and before generating items after its death. We simply don't have the required condition implemented in the code to check for that.
It was possible to add this check when burner's death effect was hardcoded, but right now it's jsonified, and we're forced to face numerous issues because of that (#56148 is another example of issue caused by monster's death effect jsonification).

**Comment by DoctorBoomstick** [comment]
So ladies and gentlemen, what I’m gleaning thus far is that, at this moment in time with the code capabilities we’ve got, the only practical solutions we have access to fixing these problems are,
1.	Updating the burner’s description to reflect that it’s got a miner chance to detonate.
2.	Add a strap to the flamer to show that it’s a backpack style unit.
3.	Causing the flamer to spawn damaged, cause a fully intact flamethrower isn’t going to be leaking napalm, and it’s definitely not going to survive, you know, blowing up.
4.	Adjusting the amount of napalm the flamer can spawn with.
At the risk of shameless self advertisement, two of these issues are addressed potentially by #59296.

**Comment by Dan1ss1mo** [comment]
On the simular note, should zomborgs have some hints about them exploding? Imo it is even less intuitive then burners
